executor/brie.go

```diff
@@ -320,41 +551,80 @@ func (e *BRIEExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			select {
 			case <-ticker.C:
 				if atomic.LoadUint32(&e.ctx.GetSessionVars().Killed) == 1 {
-					bq.cancelTask(taskID)
+					bq.cancelTask(ctx, taskID, sqlExecutor, e.info.kind != ast.BRIEKindImport)
+					item.cancel()
 					return
 				}
+				glue.Flush(taskCtx, taskID)
 			case <-taskCtx.Done():
 				return
 			}
 		}
 	}()
 
-	progress, err := bq.acquireTask(taskCtx, taskID)
+	err = bq.acquireTask(taskCtx, taskID, sqlExecutor)
 	if err != nil {
 		return err
 	}
 	defer bq.releaseTask()
 
-	e.info.execTime = types.CurrentTime(mysql.TypeDatetime)
-	glue := &tidbGlueSession{se: e.ctx, progress: progress, info: e.info}
+	e.info.protected.Lock()
+	e.info.protected.execTime = types.CurrentTime(mysql.TypeDatetime)
+	e.info.protected.Unlock()
 
 	switch e.info.kind {
 	case ast.BRIEKindBackup:
-		err = handleBRIEError(task.RunBackup(taskCtx, glue, "Backup", e.backupCfg), ErrBRIEBackupFailed)
+		err = handleBRIEError(brtask.RunBackup(taskCtx, glue, "Backup", e.backupCfg), ErrBRIEBackupFailed)
 	case ast.BRIEKindRestore:
-		err = handleBRIEError(task.RunRestore(taskCtx, glue, "Restore", e.restoreCfg), ErrBRIERestoreFailed)
+		err = handleBRIEError(brtask.RunRestore(taskCtx, glue, "Restore", e.restoreCfg), ErrBRIERestoreFailed)
+	case ast.BRIEKindImport:
+		l := lightning.New(e.importGlobalCfg)
+		e.importTaskCfg.Checkpoint.Schema += strconv.FormatUint(taskID, 10)
+		e.importTaskCfg.TikvImporter.SortedKVDir += strconv.FormatUint(taskID, 10)
+		err2 := handleBRIEError(l.RunOnce(taskCtx, e.importTaskCfg, glue, logutil.Logger(ctx)), ErrBRIEImportFailed)
+		if err2 != nil {
+			e.info.protected.Lock()
+			e.info.protected.message = err2.Error()
```

Comment:
Maybe we can define some member functions for `brieTaskInfo` like `SetMessage`, `GetMessage` to make the code more readable.

And we can define some SetXX, GetXX functions for brieTaskProgress as well